### PR TITLE
[FIX] account: Canceled journal entries still exist as analytic entries

### PR DESCRIPTION
**Pull request by agr-odoo**
Create a Journal entry
On the lines add an analytic account
Save, Post and Cancel

Even if the move is now back to draft state the analytic line still
exists under Avdiser>Analytic entries

opw-2300192

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by agr-odoo** [comment]
@fw-bot ignore

**Comment by agr-odoo** [comment]
@nim-odoo

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -180,6 +180,7 @@ def button_cancel(self):
                        'SET state=%s '\
                        'WHERE id IN %s', ('draft', tuple(self.ids),))
             self.invalidate_cache()
+            self.mapped('line_ids').mapped('analytic_line_ids').unlink()
```
Comment (nim-odoo) [inline addons/account/models/account_move.py:183]:
```suggestion
            self.mapped('line_ids.analytic_line_ids').unlink()
```

**Comment by agr-odoo** [comment]
> The fix seems logical, but why no forward-port?

checked in runbot v12 and it's already fixed.
Plus, it was fixed with
https://github.com/odoo/odoo/blob/2298097eb866b786c8e902402bf72bb0521f34c1/addons/account/models/account_move.py#L366
so I'll stick with the same code

**Pull request closed without merging**